### DISCUSSION TITLE
为什么reality配置失败？

### DISCUSSION THREAD

**Discussion by wangmice** [XTLS/Xray-core#2003] [General]
客户端和服务端都是1.8.1
服务端配置如下：关键配置处
"security": "reality",
                  "realitySettings": {
                      "show": true,
                      "dest": "80",
                      "xver": 0,
                      "serverNames": ["服务器域名"],
                      "privateKey": "XXXXXXXXXXXXXXXXXXXXXXXXXXX",
                      "shortIds": [
                           "",
                           "0123",
                           "01234567",
                           "0123456789abcdef"
                      ]
                   },
                       "sniffing": {
                           "enabled": true,
                           "destOverride": [
                             "http",
                             "tls"
                           ]
                       }

服务端日志文件没有错误输出，以下是服务端systemd journal
hs.c.AuthKey[:16]: [91 65 19 202 175 15 62 209 21>
REALITY remoteAddr: 客户端IP:端口       hs.c.ClientVer: [1 8 1]
REALITY remoteAddr: 客户端IP:端口        hs.c.ClientTime: 2023-04-29 11:49:43 +0800 CST
REALITY remoteAddr: 客户端IP:端口       hs.c.ClientShortId: [254 35 0 0 0 0 0 0]
REALITY remoteAddr: 客户端IP:端口        hs.c.conn == conn: true
REALITY remoteAddr: 客户端IP:端口        handled: false

客户端配置：
                "security": "reality",
                 "realitySettings": {
                     "show": true,
                     "fingerprint": "randomized",
                     "serverName": "服务器域名",
                     "publicKey": "XXXXXXXXXXXXXXXXXXX",
                     "shortId": "0123",
                     "spiderX": "/robot"
                 },
  
日志如下：

2023/04/29 11:50:51 [Warning] [711696793] app/proxyman/outbound: failed to process outbound traffic > proxy/vless/outbound: failed to find an available destination > common/retry: [EOF dial tcp 服务器IP: operation was canceled] > common/retry: all retry attempts failed
2023/04/29 11:50:51 [Warning] [2271365908] app/proxyman/outbound: failed to process outbound traffic > proxy/vless/outbound: failed to find an available destination > common/retry: [EOF dial tcp 服务器IP: operation was canceled] > common/retry: all retry attempts failed
2023/04/29 11:50:51 [Warning] [2730667776] app/proxyman/outbound: failed to process outbound traffic > proxy/vless/outbound: failed to find an available destination > common/retry: [EOF dial tcp 服务器IP: operation was canceled] > common/retry: all retry attempts failed
2023/04/29 11:50:51 [Warning] [4022686976] app/proxyman/outbound: failed to process outbound traffic > proxy/vless/outbound: failed to find an available destination > common/retry: [EOF dial tcp 服务器IP: operation was canceled] > common/retry: all retry attempts failed
2023/04/29 11:50:52 [Warning] app/proxyman/outbound: failed to process outbound traffic > proxy/vless/outbound: failed to find an available destination > common/retry: [EOF] > common/retry: all retry attempts failed

wget -O-  -4 https://www.facebook.com

正在解析主机 www.facebook.com (www.facebook.com)... 31.13.70.36
正在连接 www.facebook.com (www.facebook.com)|31.13.70.36|:443... 已连接。
OpenSSL: error:0A000126:SSL routines::unexpected eof while reading
无法建立 SSL 连接。

**Comment by wangmice** [comment]
配置成tls无异常，就reality有问题，完全是按照示例上配置的

**Comment by wangmice** [comment]
用chrome显示：
无法访问此网站， 意外终止了连接。
请试试以下办法：

检查网络连接
[检查代理服务器和防火墙](chrome-error://chromewebdata/#buttons)
[运行 Windows 网络诊断](javascript:diagnoseErrors())
ERR_CONNECTION_CLOSED

**Comment by chika0801** [comment]
建议多换些其它目标网址。或使用 https://github.com/chika0801/Xray-examples/tree/main/VLESS-XTLS-uTLS-REALITY/any_SNI_No_SNI 这个能通。

不觉得麻烦自己偷自己肯定通。

类似现象我遇到过，没研究出原因。
